### PR TITLE
Add public IP option to launch configuration

### PR DESCRIPTION
**Pull request by RichardWLaub**
When deploying worker nodes to private subnets I wanted the option to disable public IPs. This is my first code contribution to a pulumi project so any feedback is appreciated!

### CODE REVIEW

File: nodejs/eks/cluster.ts [hunk 1]
```diff
@@ -479,7 +485,7 @@ ${customUserData}
         }
 
         const nodeLaunchConfiguration = new aws.ec2.LaunchConfiguration(`${name}-nodeLaunchConfiguration`, {
-            associatePublicIpAddress: true,
+            associatePublicIpAddress: args.associatePublicIpAddress && true,
```
Comment (lukehoban) [inline nodejs/eks/cluster.ts]: This should be `|| true` (like in the `instanceType` case below).

**Comment by RichardWLaub** [comment]
```
import * as awsinfra from "@pulumi/aws-infra";
import * as eks from "@pulumi/eks";

const prefix = "richardtest";
const numAvailabilityZones = 3;

const network = new awsinfra.Network(`${prefix}-net`, {
    numberOfAvailabilityZones: numAvailabilityZones, // Create subnets in many AZs
    usePrivateSubnets: true,                         // Run compute inside private subnets in each AZ
});

const cluster = new eks.Cluster("cluster", {
    deployDashboard: false,
    desiredCapacity: 4,
    maxSize: 4,
    subnetIds: network.subnetIds,
    vpcId: network.vpcId,
});
```
@lukehoban This was the sample code that was not working before my changes. Basically I expected the worker nodes to be deployed to the private subnets without public IPs assigned. The nodes were deployed to the private subnets but they were assigned public IP addresses anyway. 

Like you mentioned there is probably a better way to set a default using `computeWorkerSubnets` instead. I'll look into that when I have cycles. Thanks for the review!

**Comment by lukehoban** [comment]
Great - yes - that code sample is exactly what I was imagining we should add to test coverage.  There is some code specifically to handle private subnets in the current implementation, but sounds like they don't work correctly currently?  

We may want to have an explicit `usePrivateSubnets` option just to be clear about the intention instead of just inferring it from the subnets passed in. @pgavlin probably has thoughts on this as he added the logic here to do the inference.

**Comment by metral** [comment]
https://github.com/pulumi/pulumi-eks/pull/81 resolved this issue, so closing the PR.

**Pull request closed without merging**